crates/modelardb_server/src/storage/uncompressed_data_buffer.rs

```diff
@@ -198,14 +199,31 @@ impl fmt::Debug for UncompressedInMemoryDataBuffer {
 
 #[async_trait]
 impl UncompressedDataBuffer for UncompressedInMemoryDataBuffer {
-    /// Finish the array builders and return the data in a structured [`RecordBatch`].
+    /// Finish the array builders and return the data in a [`RecordBatch`] sorted by time.
     async fn record_batch(&mut self) -> Result<RecordBatch, ParquetError> {
-        let timestamps = self.timestamps.finish();
+        // SortColumn requires that the values are in an Arc but this is not needed for take.
+        let timestamps = Arc::new(self.timestamps.finish());
         let values = self.values.finish();
 
+        // sort_to_indices is used instead of lexsort as it is unclear how lexsort sorts multiple
+        // arrays, instead the same combination of sort_to_indices and take used in lexsort is used.
+        // unwrap() is safe as only supported types are sorted and indices cannot be out of bounds.
+        let sort_indices = compute::lexsort_to_indices(
```

Comment:
Maybe change to `sorted_indices` for consistency.

Reply:
Fixed.